wicamo96/FieldScout
# Crud fixes

# Description

Tweaked how things are deleted to be sure all info associated with the id of interest. 

## Type of change


# Testing Instructions

### Create database and populate data if necessary
- If you haven't created the FieldScout database, please do so in visual studio community.
- Use this to fill out tables and columns: [SQL data](https://github.com/wicamo96/FieldScout/blob/main/SQL/01_FieldScout_Create_DB.sql)
- Use this to seed the database: [SQL Seed Data](https://github.com/wicamo96/FieldScout/blob/main/SQL/02_FieldScout_Seed_Data.sql)

### Testing CRUD
- Click the execute with debugger solid green button to launch the backend in visual studio community.
- From the client directory in a git terminal, run `npm run dev`.
- Once the swagger window appears, open a new tab and navigate to `http://localhost:5173/`
- You should be directed to the log in page.
- Log in with the default email.
- Once logged in, you will be taken to the dashboard page 
- Navigate to the greenhouse management menu and create test data for house, bays, and bay divisions
- Test updating information and deleting information from different levels, If you delete a bay or house that has bay/bay division information associated with it, you shouldn't receive any errors and the info should all be deleted.
- Navigate to the pest management menu
- Enter a test pest and ensure you're able to also update and delete the pest.
- Navigate to the scouting menu and enter test data for one of the bays in one of the houses
- You should be able to add, edit, and delete scouting data without errors.

# Checklist:
